# Integral

In [1]:
import sympy as sy

### BY SYMBOLIC

In [2]:
x = sy.Symbol('x')
f = 2*x**2 + 4*x -1
f

2*x**2 + 4*x - 1

In [3]:
sy.integrate(f)

2*x**3/3 + 2*x**2 - x

### NUMERIK

Numerik Fungsi 
- Ini kalo punya fungsinya tapi ingin menghitung integral berupa numerik yang outputnya berupa angka

In [4]:
import scipy as scipy
from scipy import integrate

In [6]:
def f(x):
    return 2*x**2 + 4*x -1

A = integrate.quad(f,2,5) #ini f nama fungsinya. 2 ama 5 itu batas bawah ama batas atas
print(A)

(117.0, 1.2989609388114332e-12)


- nilai integralnya adalah yang kiri (yg 117), yg kanan yg aneh itu dia nilai error. 
- error didapatkan karena ini kan perhitungan numerik.

Numerik Data/Array

In [7]:
from scipy import integrate
import numpy as np

In [8]:
def f(x):
    return 2*x**2 + 4*x -1

x = np.linspace(2,5)
y = f(x)

A = integrate.trapz(y, x)
A

117.00374843815077

1. kalo integrate.quad inputannya fungsi
2. kalo integrate.trapz inputnya adalah data/array

In [9]:
#Case Menghitung Total Pendapatan 2019 di Supermarket Cabang Yangon
import pandas as pd

In [10]:
data = pd.read_csv('https://raw.githubusercontent.com/fahmimnalfrzki/ds-masterclass-h8-vol2/main/data/supermarket_sales%20-%20Sheet1.csv')

In [11]:
data.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

In [13]:
data = data[data['City']== 'Yangon'].sort_values(['Date', 'Time']).reset_index(drop=True)

In [14]:
data.head()

Invoice ID Branch    City Customer type  Gender            Product line  \
0  765-26-6951      A  Yangon        Normal    Male       Sports and travel   
1  651-88-7328      A  Yangon        Normal  Female     Fashion accessories   
2  416-17-9926      A  Yangon        Member  Female  Electronic accessories   
3  530-90-9855      A  Yangon        Member    Male      Home and lifestyle   
4  182-52-7000      A  Yangon        Member  Female       Sports and travel   

   Unit price  Quantity  Tax 5%    Total      Date   Time      Payment  \
0       72.61         6  21.783  457.443  1/1/2019  10:39  Credit card   
1       65.74         9  29.583  621.243  1/1/2019  13:55         Cash   
2       74.22        10  37.110  779.310  1/1/2019  14:42  Credit card   
3       47.59         8  19.036  399.756  1/1/2019  14:47         Cash   
4       27.04         4   5.408  113.568  1/1/2019  20:26      Ewallet   

     cogs  gross margin percentage  gross income  Rating  
0  435.66                 4.761905        21.783     6.9  
1  591.66                 4.761905        29.583     7.7  
2  742.20                 4.761905        37.110     4.3  
3  380.72                 4.761905        19.036     5.7  
4  108.16                 4.761905         5.408     6.9